hashes/src/sha256.rs

```diff
@@ -155,16 +155,17 @@ impl Midstate {
 
     /// Copies a byte slice into the [`Midstate`] object.
     pub fn from_slice(sl: &[u8]) -> Result<Midstate, Error> {
-        if sl.len() != Self::LEN {
-            Err(Error::InvalidLength(Self::LEN, sl.len()))
-        } else {
+        if sl.len() == Self::LEN {
             let mut ret = [0; 32];
             ret.copy_from_slice(sl);
             Ok(Midstate(ret))
+        } else {
+            Err(Error::InvalidLength(Self::LEN, sl.len()))
```

Comment:
I prefer the "early return" order of this code.

Reply:
I prefer `Ok(Midstate(sl.try_into().map_err(|_| Error::InvalidLength(Self::LEN, sl.len()))?))`

Reply:
ha! point goes to @Kixunil, jokes aside, I won't rearrange any early returns in future (I"ll split this PR up and resolve this comment while doing so).